parquet-variant-compute/src/variant_get/output/variant.rs

```diff
@@ -44,40 +85,103 @@ impl OutputBuilder for VariantOutputBuilder<'_> {
         _value_field: &BinaryViewArray,
         typed_value: &ArrayRef,
     ) -> arrow::error::Result<ArrayRef> {
-        // in this case dispatch on the typed_value and
-        // TODO macro'ize this using downcast! to handle all other primitive types
         // TODO(perf): avoid builders entirely (and write the raw variant directly as we know the metadata is the same)
-        let mut array_builder = VariantArrayBuilder::new(variant_array.len());
         match typed_value.data_type() {
+            DataType::Int8 => {
+                cast_partially_shredded_primitive!(
+                    typed_value,
+                    variant_array,
+                    Int8Type,
+                    DataType::Int8
+                )
+            }
```

Comment:
aside: I'm a bit surprised that `fmt` didn't simplify these match arms:
```suggestion
            DataType::Int8 => cast_partially_shredded_primitive!(
                typed_value,
                variant_array,
                Int8Type,
                DataType::Int8
            ),
```

Reply:
I wonder if it's having trouble inferring things across the macro boundary, so it doesn't know that it can simplify the match arms. I've noticed that macros sometimes behave strangely when it comes to formatting(and LSP diagnostics).

Reply:
I actually suspect `fmt` just plain isn't running, for some reason? I've seen other places in this file that aren't formatted as I would have expected.